File: android_audio_video/app/src/main/java/com/example/myapplicationl4/MainActivity.java
package com.example.myapplicationl4;

import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Intent intent_audio;
    private Intent intent_video;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void Video(View view) {
        if(intent_video == null)
            intent_video = new Intent(this, VideoActivity.class);

        startActivity(intent_video);
    }

    public void Audio(View view) {
        if(intent_audio == null)
            intent_audio = new Intent(this, AudioActivity.class);

        startActivity(intent_audio);

    }

}

File: lab1/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.text.method.HideReturnsTransformationMethod
import android.text.method.PasswordTransformationMethod
import android.widget.*
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val button = findViewById<Button>(R.id.Btn)
        val et = findViewById<EditText>(R.id.pwd)
        val tv = findViewById<TextView>(R.id.textView)
        val rg = findViewById<RadioGroup>(R.id.radiogroup)


        rg.setOnCheckedChangeListener { _, checkedId ->
            when (checkedId) {
                R.id.radiobutton1 -> et.transformationMethod = HideReturnsTransformationMethod.getInstance()
                R.id.radiobutton2 -> et.transformationMethod = PasswordTransformationMethod.getInstance()

            }
        }

        button.setOnClickListener{
            tv.text = ""
            if(et.text.toString().isNotEmpty())
            tv.text = et.text
            else{
                val myToast = Toast.makeText(this, "Заполните поле пароля", Toast.LENGTH_SHORT)
                myToast.show()
            }
        }
    }
}
